ygot/struct_validation_map.go

```diff
@@ -332,18 +332,35 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 	return o, nil
 }
 
+// MergeOption is an interface that is implemented by all parameters which specify a merge
+// behaviour when merging GoStructs.
+type MergeOption interface {
+	// IsMergeOption is a marker method for these options.
+	IsMergeOption()
+}
+
+// AllowLeafOverwrite allows scalar (leaf) fields to be overwritten in the destination
+// strucft when specified.
+type AllowLeafOverwrite struct {
+	IfEqual bool // IfEqual allows leaves to be overwritten only if their value is equal.
+}
+
+// IsMergeOption marks AllowLeafOverwrite as a valid MergeOption.
+func (*AllowLeafOverwrite) IsMergeOption() {}
+
+// NewAllowLeafOverwrite returns a new AllowLeafOverwrite pointer. The ifeq argument
+// specifies whether merging should only be allowed if the source and destination
+// leaves have equal values.
+func NewAllowLeafOverwrite(ifeq bool) *AllowLeafOverwrite { return &AllowLeafOverwrite{IfEqual: ifeq} }
+
 // MergeStructs takes two input ValidatedGoStructs and merges their contents,
 // returning a new ValidatedGoStruct. If the input structs a and b are of
 // different types, an error is returned.
 //
-// In the case that the structs contain a slice, or a map that is already
-// populated in both structs, an error is returned. Merging two lists with
-// identical members will be added in future iterations of this code.
-//
-// TODO(robjs): Fix the unimplemented test cases where two structs of
-// the same type have slices or maps that are already populated.
-// See https://github.com/openconfig/ygot/issues/74.
-func MergeStructs(a, b ValidatedGoStruct) (ValidatedGoStruct, error) {
+// Where two structs contain maps or slices that are populated in both a and b
+// their contents are merged. If a leaf is populated in both a and b, an error
+// is returned unless the AllowLeafOverwrite MergeOption is supplied.
+func MergeStructs(a, b ValidatedGoStruct, opts ...MergeOption) (ValidatedGoStruct, error) {
```

Comment:
b overwrites a should be explicitly stated.

Reply:
Done.